torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -988,13 +988,162 @@ const Expr* PolynomialTransformer::mutate(const Div* v) {
     return getImmediateByType(v->dtype(), 1);
   }
 
+  auto* lhsTerm = dynamic_cast<const Term*>(lhs_new);
+  auto* rhsTerm = dynamic_cast<const Term*>(rhs_new);
+
+  if (const Polynomial* lhsPoly = dynamic_cast<const Polynomial*>(lhs_new)) {
+    if (auto* term = factorizePolynomial(lhsPoly)) {
+      lhsTerm = term;
+    }
+  }
+  if (const Polynomial* rhsPoly = dynamic_cast<const Polynomial*>(rhs_new)) {
+    if (auto* term = factorizePolynomial(rhsPoly)) {
+      rhsTerm = term;
+    }
+  }
+
+  if (lhsTerm && !rhsTerm) {
+    rhsTerm =
+        new Term(hasher_, getImmediateByType(rhs_new->dtype(), 1), {rhs_new});
+  } else if (rhsTerm && !lhsTerm) {
+    lhsTerm =
+        new Term(hasher_, getImmediateByType(lhs_new->dtype(), 1), {lhs_new});
+  }
+
+  if (lhsTerm && rhsTerm) {
+    auto lVars = lhsTerm->variables();
+    auto rVars = rhsTerm->variables();
+    bool didAnything = false;
+
+    // we want to find the difference between these vectors, which are sorted by
+    // hash.
+    for (auto lIt = lVars.begin(); lIt != lVars.end() && !rVars.empty();) {
```

Comment:
This entire loop is tricky, especially given its use of iterators. It deserves its own method.

Reply:
I don't see the resolution here.